src/plasmapy/simulation/particle_tracker/particle_tracker.py

```diff
@@ -620,117 +703,205 @@ def particles_on_grid(self):
+                    field_value,
+                    0.0 * AbstractGrid.recognized_quantities[field_name].unit,
+                )
 
         # Create arrays of E and B as required by push algorithm
         E = np.array(
-            [Ex.to(u.V / u.m).value, Ey.to(u.V / u.m).value, Ez.to(u.V / u.m).value]
+            [
+                summed_field_values["E_x"].to(u.V / u.m).value,
+                summed_field_values["E_y"].to(u.V / u.m).value,
+                summed_field_values["E_z"].to(u.V / u.m).value,
+            ]
         )
         E = np.moveaxis(E, 0, -1)
-        B = np.array([Bx.to(u.T).value, By.to(u.T).value, Bz.to(u.T).value])
+        B = np.array(
+            [
+                summed_field_values["B_x"].to(u.T).value,
+                summed_field_values["B_y"].to(u.T).value,
+                summed_field_values["B_z"].to(u.T).value,
+            ]
+        )
         B = np.moveaxis(B, 0, -1)
 
+        return summed_field_values, E, B
+
+    def _update_time(self, summed_field_values):
         # Calculate the adaptive time step from the fields currently experienced
         # by the particles
         # If user sets dt explicitly, that's handled in _adaptive_dt
         if self._is_adaptive_time_step:
-            dt = self._adaptive_dt(Ex, Ey, Ez, Bx, By, Bz)
+            dt = self._adaptive_dt(
+                summed_field_values["E_x"],
+                summed_field_values["E_y"],
+                summed_field_values["E_z"],
+                summed_field_values["B_x"],
+                summed_field_values["B_y"],
+                summed_field_values["B_z"],
+            )
         else:
             dt = self.dt
 
-        # Check if the beta threshold has been achieved if the simulation is not
-        # already using the relativistic integrator
-        if not self._integrator.is_relativistic:
-            beta = self.vmax / const.c.si.value
-
-            if beta >= self._beta_threshold:
-                self._integrator = RelativisticBorisIntegrator()
-
         # Make sure the time step can be multiplied by a [nparticles, 3] shape field array
         if isinstance(dt, np.ndarray) and dt.size > 1:
-            dt = dt[tracked_mask, np.newaxis]
+            dt = dt[self._tracked_particle_mask, np.newaxis]
 
             # Increment the tracked particles' time by dt
-            self.time[tracked_mask] += dt
+            self.time[self._tracked_particle_mask] += dt
         else:
             self.time += dt
 
-        # Update the tracked particles using the integrator specified at instantiation
-        # TODO: implement "tentative" x and v to prevent speeds faster than light
-        #  from occurring over one step. Possibly based on field strengths?
-        self.x[tracked_mask], self.v[tracked_mask] = self._integrator.push(
-            pos_tracked, vel_tracked, B, E, self.q, self.m, dt
+        return dt
+
+    def _update_position(self, B, E):
```

Comment:
Add docstrings for all these submethods  - just a single sentence defining what it does.